autotest/src/main/kotlin/no/nav/familie/tilbake/e2e/domene/builder/KravgrunnlagBuilder.kt

```diff
@@ -0,0 +1,171 @@
+            this.vedtakId = vedtakId ?: Random.nextInt(100000, 999999).toBigInteger()
+            this.kodeStatusKrav = status.toString()
+            this.kodeFagomraade = utledFagområdeKode(ytelsestype = ytelsestype)
+            this.fagsystemId = eksternFagsakId
+            this.vedtakIdOmgjort = BigInteger.ZERO
+            this.vedtakGjelderId = "12345678901"
+            this.typeGjelderId = TypeGjelderDto.PERSON
+            this.utbetalesTilId = "12345678901"
+            this.typeUtbetId = TypeGjelderDto.PERSON
+            this.enhetAnsvarlig = "8020"
+            this.enhetBosted = "8020"
+            this.enhetBehandl = "8020"
+            this.kontrollfelt = finalKontrollfelt
+            this.saksbehId = "K231B433"
+            this.referanse = eksternBehandlingId
+            this.tilbakekrevingsPeriode.addAll(tilbakekrevingsPerioder(antallPerioder = antallPerioder,
+                                                                       under4rettsgebyr = under4rettsgebyr,
+                                                                       muligforeldelse = muligforeldelse,
+                                                                       ytelsestype = ytelsestype,
+                                                                       periodeLengde = periodeLengde))
+        }
+    }
+
+    private fun utledFagområdeKode(ytelsestype: Ytelsestype): String {
+        return when (ytelsestype) {
+            Ytelsestype.BARNETRYGD -> "BA"
+            Ytelsestype.OVERGANGSSTØNAD -> "EFOG"
+            Ytelsestype.BARNETILSYN -> "EFBT"
+            Ytelsestype.SKOLEPENGER -> "EFSP"
+            Ytelsestype.KONTANTSTØTTE -> "KS"
+        }
+    }
+
+    private fun tilbakekrevingsPerioder(antallPerioder: Int,
+                                        under4rettsgebyr: Boolean,
+                                        muligforeldelse: Boolean,
+                                        ytelsestype: Ytelsestype,
+                                        periodeLengde: Int, ): List<DetaljertKravgrunnlagPeriodeDto> {
+        /*Lager alltid periodene 3 måneder lange med 1 måned mellom, så derfor gange med 4.
+        Første periode starter for 3 år siden når det skal være muligforeldelse eller for 4 måneder siden gange med antall perioder*/
+        var startDato = if (muligforeldelse) {
+            val antallAarTilbakeITid = if (periodeLengde > 3) 4L else 3L
+            LocalDate.now().minusYears(antallAarTilbakeITid).withDayOfMonth(1)
+        } else {
+            val multiplierPgaPeriodeLengde = if (periodeLengde > 3) 2 else 1
+            LocalDate.now()
+                    .minusMonths(BigDecimal(4).multiply(BigDecimal((antallPerioder * multiplierPgaPeriodeLengde))).toLong())
+                    .withDayOfMonth(1)
+        }
+
+        /*Setter feilutbetalt beløp til 20000 kr, med mindre det skal være under4rettsgebyr, da skal det være under 4796 kr pr jan.2021*/
```

Comment:
kommentar er for lang (mer enn 130 tegn)

Reply:
Endret kommentaren